src/snowcli/cli/constants.py

```diff
@@ -1,24 +1,36 @@
 from __future__ import annotations
 
+from dataclasses import dataclass
 from enum import Enum
 
 DEPLOYMENT_STAGE = "deployments"
 
 
+@dataclass(frozen=True)
+class ObjectNames:
+    cli_name: str
+    sf_name: str
+    sf_plural_name: str
+
+
 class ObjectType(Enum):
-    COMPUTE_POOL = "compute pool"
-    DATABASE = "database"
-    FUNCTION = "function"
-    JOB = "job"
-    PROCEDURE = "procedure"
-    ROLE = "role"
-    SCHEMA = "schema"
-    SERVICE = "service"
-    STREAMLIT = "streamlit"
-    TABLE = "table"
-    WAREHOUSE = "warehouse"
+    COMPUTE_POOL = ObjectNames("compute-pool", "compute pool", "compute pools")
+    DATABASE = ObjectNames("database", "database", "databases")
+    FUNCTION = ObjectNames("function", "function", "functions")
+    JOB = ObjectNames("job", "job", "jobs")
+    PROCEDURE = ObjectNames("procedure", "procedure", "procedures")
+    ROLE = ObjectNames("role", "role", "roles")
+    SCHEMA = ObjectNames("schema", "schema", "schemas")
+    SERVICE = ObjectNames("service", "service", "services")
+    STAGE = ObjectNames("stage", "stage", "stages")
+    STREAMLIT = ObjectNames("streamlit", "streamlit", "streamlits")
+    TABLE = ObjectNames("table", "table", "tables")
+    WAREHOUSE = ObjectNames("warehouse", "warehouse", "warehouses")
+
+
+OBJECT_TO_NAMES = {o.value.cli_name: o.value for o in ObjectType}
 
 
 class SnowparkObjectType(Enum):
-    PROCEDURE = ObjectType.PROCEDURE.value
-    FUNCTION = ObjectType.FUNCTION.value
+    PROCEDURE = ObjectType.PROCEDURE.value.sf_name
```

Comment:
IMO this Enum should have also ObjectNames as values.

Reply:
Unless you want to use it for choices in typer... Refactored it, the enum is used only for execute commands so I limited the scope